File: fizz-buzz.py
# fizz-buzz.py
# Version 20160915
# Written by Harry Wong (RedAndBlueEraser)

def get_fizz_buzz(len):
    fizz_buzz = ""    # Fizz buzz sequence string.

    for i in range(1, len + 1):
        # Iteration is divisible by 3; add "Fizz".
        if i % 3 == 0:
            fizz_buzz += "Fizz"
            # Iteration is divisible by 5; add "Buzz" after "Fizz".
            if i % 5 == 0:
                fizz_buzz += " Buzz"

        # Iteration is divisible by 5; add "Buzz".
        elif i % 5 == 0:
            fizz_buzz += "Buzz"

        # Add iteration.
        else:
            fizz_buzz += str(i)

        # Separate values with a comma except for last value.
        if i != len:
            fizz_buzz += ", "

    return fizz_buzz

print get_fizz_buzz(50)

File: fizz-buzz.c
/*
 * fizz-buzz.c
 * Version 20160915
 * Written by Harry Wong (RedAndBlueEraser)
 */

#include <stdio.h>

#define FIZZ_BUZZ_MAX_LENGTH 50

/* Prints the fizz buzz sequence of specified length to stdout. */
void print_fizz_buzz(int length);

int main(int argc, char *argv[]) {
    print_fizz_buzz(FIZZ_BUZZ_MAX_LENGTH);
    printf("\n");
    return 0;
}

void print_fizz_buzz(int length) {
    /* Iteration over fizz buzz sequence. */
    int i;

    /* Iterate over fizz buzz sequence. */
    for (i = 1; i <= length; i++) {
        /* Iteration is divisible by 3; print "Fizz". */
        if (i % 3 == 0) {
            printf("Fizz");
            /* Iteration is divisible by 3 and 5; print "Buzz" after "Fizz". */
            if (i % 5 == 0) {
                printf(" Buzz");
            }
        }

        /* Iteration is divisible by 5; print "Buzz". */
        else if (i % 5 == 0) {
            printf("Buzz");
        }

        /* Print iteration. */
        else {
            printf("%d", i);
        }

        /* Separate values with a comma except for last value. */
        if (i != length) {
            printf(", ");
        }
    }
}
